examples/different-input-types/different-input-types.cxx

```diff
@@ -0,0 +1,117 @@
+
+template<typename TImage, typename TLabelImage>
+int
+OverlayLabelMap(itk::wasm::Pipeline & pipeline, const TImage * inputImage, const TLabelImage * labelImage)
+{
+  using ImageType = TImage;
+  using LabelImageType = TLabelImage;
+  constexpr unsigned int Dimension = ImageType::ImageDimension;
+
+  using LabelType = typename LabelImageType::PixelType;
+  using LabelObjectType = itk::LabelObject<LabelType, Dimension>;
+  using LabelMapType = itk::LabelMap<LabelObjectType>;
+
+  using ConverterType = itk::LabelImageToLabelMapFilter<LabelImageType, LabelMapType>;
+  using FilterType = itk::LabelMapOverlayImageFilter<LabelMapType, ImageType>;
+
+  pipeline.get_option("input-image")->required()->type_name("INPUT_IMAGE");
+  pipeline.get_option("label-image")->required()->type_name("INPUT_IMAGE");
+
+  using OutputImageType = itk::wasm::OutputImage<typename FilterType::OutputImageType>;
+  OutputImageType outputImage;
+  pipeline.add_option("output-image", outputImage, "The output image")->required()->type_name("OUTPUT_IMAGE");
+
+  ITK_WASM_PARSE(pipeline);
+
+  auto converter = ConverterType::New();
+  converter->SetInput(labelImage);
+
+  auto filter = FilterType::New();
+  filter->SetInput(converter->GetOutput());
+  filter->SetFeatureImage(inputImage);
+  filter->SetOpacity(0.5);
+
+  filter->UpdateLargestPossibleRegion();
+
+  outputImage.Set(filter->GetOutput());
+
+  return EXIT_SUCCESS;
+}
+
+template<typename TImage>
+class InputImagePipelineFunctor
+{
+public:
+  int operator()(itk::wasm::Pipeline & pipeline)
+  {
+    using InputImageType = itk::wasm::InputImage<TImage>;
+    InputImageType inputImage;
+    pipeline.add_option("input-image", inputImage, "The input image");
+
+    ITK_WASM_PRE_PARSE(pipeline);
+
+    typename TImage::ConstPointer image = inputImage.Get();
+    parsedImage = image;
+    return itk::wasm::SupportInputImageTypes<LabelImagePipelineFunctor,
+      uint8_t>
+      ::template Dimensions<2U>("label-image", pipeline);
+  }
+
+private:
+  template<typename TLabelImage>
+  class LabelImagePipelineFunctor
+  {
+  public:
+    int operator()(itk::wasm::Pipeline & pipeline)
+    {
+      using LabelImageType = itk::wasm::InputImage<TLabelImage>;
+      LabelImageType labelImage;
+      pipeline.add_option("label-image", labelImage, "The label image");
+
+      ITK_WASM_PRE_PARSE(pipeline);
+
+      typename TLabelImage::ConstPointer label = labelImage.Get();
+      return OverlayLabelMap<TImage, TLabelImage>(pipeline, parsedImage, label);
+    }
+  };
+
+  static inline const TImage * parsedImage;
+};
+
+int main( int argc, char * argv[] )
+{
+  // Create the pipeline for parsing arguments. Provide a description.
+  itk::wasm::Pipeline pipeline("different-input-types", "An itk-wasm pipeline example that demonstrates accepting different input types", argc, argv);
+
+  return itk::wasm::SupportInputImageTypes<InputImagePipelineFunctor,
+   uint8_t,
+   uint16_t>
+  ::Dimensions<2U>("input-image", pipeline);
```

Comment:
If I add a third Dimension
`::Dimensions<2U, 3U>("input-image", pipeline);`
I get compile errors.

How can I narrow down the TImage and TLabelImage dimension to just be matching?

Reply:
Good point! Let's fix the second image dimension to the first image dimension as a best practice. Also adding .gitignore.